### PR TITLE
Move metrics poller to seperate class

### LINKED ISSUE #104: Stop polling when 401 is detected
screenshot from 2018-01-23 09-39-04

### PR DESCRIPTION
**Pull request by ivarconr**
Instead of having various components starting a metrics poller we can move this logic to a specialized class handling an in background poller. We should not fetch metrics before we know that there exists any toggles. 

Todo:
- [x] Write unit tests

This closes #104

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -34,6 +35,7 @@ if (process.env.NODE_ENV !== 'production' && window.__REDUX_DEVTOOLS_EXTENSION_C
 }
 
 const unleashStore = createStore(store, composeEnhancers(applyMiddleware(thunkMiddleware)));
+new MetricsPoller(unleashStore);
```
Comment (SimenB) [inline src/index.jsx]:
```
  38:1  error  Do not use 'new' for side effects  no-new
```

Move `start` out of the constructor and call it explicitly?

File: src/metrics-poller.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.timer = undefined;
+        this.running = false;
+
+        this.start();
+    }
+
+    start() {
+        const store = this.store;
+        store.subscribe(() => {
+            const features = store.getState().features.toJS();
```
Comment (SimenB) [inline src/metrics-poller.js]: is the `toJS()` needed?

File: src/metrics-poller.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                this.timer = setInterval(this.fetchMetrics.bind(this), 5000);
+            }
+        });
+    }
+
+    fetchMetrics() {
+        this.store.dispatch(fetchFeatureMetrics());
+    }
+
+    destroy() {
```
Comment (SimenB) [inline src/metrics-poller.js]: is this ever called? Is it a problem if it's not?

Comment (ivarconr) [reply src/metrics-poller.js]: it is via tests. It should not be a problem. Unleash will poll metrics as long as unleash is running and there are active toggles.

**Pull request merged**